monai/inferers/merger.py

```diff
@@ -246,8 +247,20 @@ def __init__(
         self.value_dtype = value_dtype
         self.count_dtype = count_dtype
         self.store = store
-        self.value_store = zarr.storage.TempStore() if value_store is None else value_store
-        self.count_store = zarr.storage.TempStore() if count_store is None else count_store
+        if version_geq(get_package_version("zarr"), "3.0.0"):
+            if value_store is None:
+                with TemporaryDirectory() as tmpdir:
```

Comment:
`TemporaryDirectory` should delete the directory once the context block if left, is this a problem here? It might not be immediate but we can't rely on it existing outside the block I don't think. If you store an instance of `TemporaryDirectory` as an attribute it will get cleaned up when the object is collected.

Reply:
Thanks! should be an issue, updated it in the latest commit.